src/frontend/src/utils/canisterIdResolution.test.ts

```diff
@@ -0,0 +1,63 @@
+import { resolveCanisterId } from "$src/utils/canisterIdResolution";
+
+const HEADER_NAME = "x-ic-canister-id";
+
+test("should resolve canister id", async () => {
+  const fetchMock = vi.fn();
+  fetchMock.mockReturnValueOnce(
+    new Response(null, {
+      status: 200,
+      headers: [[HEADER_NAME, "bkyz2-fmaaa-aaaaa-qaaaq-cai"]],
+    })
+  );
+  global.fetch = fetchMock;
+
+  expect(
+    await resolveCanisterId({
+      origin: "https://example.com",
+    })
+  ).toEqual({ ok: "bkyz2-fmaaa-aaaaa-qaaaq-cai" });
+});
+
+test("should not resolve canister id on missing header", async () => {
+  const fetchMock = vi.fn();
+  fetchMock.mockReturnValueOnce(
+    new Response(null, {
+      status: 200,
+    })
+  );
+  global.fetch = fetchMock;
+
+  expect(
+    await resolveCanisterId({
+      origin: "https://example.com",
+    })
+  ).toEqual("not_found");
+});
+
+test("should resolve canister id from well-known domain", async () => {
+  global.fetch = vi.fn();
+  const canisterId = "bkyz2-fmaaa-aaaaa-qaaaq-cai";
+  const wellKnownDomains = [
+    "ic0.app",
+    "icp0.io",
+    "icp0.io",
```

Comment:
Why is this duplicated?

Reply:
To make the test fail in case somebody accidentally changes the list in the production code. If somebody intentionally changes the list, they will most likely also look to change the test.

Reply:
Sorry, I don't understand it. What does it help to have it twice?

Reply:
Thanks!